org/mozilla/jss/nss/PR.java

```diff
@@ -72,6 +72,18 @@ public static native PRFDProxy NewBufferPRFD(BufferProxy read_buf,
      */
     public static native int Close(PRFDProxy fd);
 
+    /**
+     * Close an existing SSLFDProxy.
+     *
+     * See also: org.mozilla.jss.nss.PR.Close
+     *           org.mozilla.jss.nss.SSLFD.releaseNativeResources
+     */
+    public static int Close(SSLFDProxy fd) throws Exception {
+        int ret = PR.Close((PRFDProxy) fd);
+        fd.close();
```

Comment:
Usually we clean up the subclass first then the super class. If we run `fd.close()` before `PR.Close(PRFDProxy)` would that cause any problem? I'll let you decide since I'm not familiar with NSS / NSPR API at this level. Regardless, the code looks fine to me, ACK.

Reply:
The first close simply closes the socket and frees resources related to the socket in the NSPR layer. The second `close(...)` calls `releaseNativeResources(...)` from `NativeProxy`. If the order is swapped, the first call will succeed, but the second will fail, since there's no pointer to the file descriptor, which means that the NSPR resources won't get freed and thus leak.